layer.go

```diff
@@ -67,46 +69,49 @@ type Layer struct {
 	// finalHandler stores the final middleware chain handler.
 	finalHandler http.Handler
 
-	// memo stores the memoized middleware call chain by specific phase.
-	memo map[string]http.Handler
+	// parent stores the parent middleware layer to use. Use SetParent(parent).
+	parent Middleware
 
-	// stack stores the plugins registered in the current middleware instance.
+	// Pool stores the phase-specific middleware handlers stack.
 	Pool Pool
 }
 
 // New creates a new middleware layer.
 func New() *Layer {
-	return &Layer{Pool: make(Pool), memo: make(map[string]http.Handler), finalHandler: FinalHandler}
+	return &Layer{Pool: make(Pool), finalHandler: FinalHandler}
 }
 
-// Flush flushes the plugins stack.
+// Flush flushes the middleware pool.
 func (s *Layer) Flush() {
-	s.Pool = Pool{}
+	s.Pool = make(Pool)
 }
 
-// Use registers a new request handler in the middleware stack.
+// Use registers new handlers for the given phase in the middleware stack.
 func (s *Layer) Use(phase string, handler ...interface{}) {
 	s.use(phase, Normal, handler...)
 }
 
-// UsePriority registers a new request handler in the middleware stack with the given priority.
+// UsePriority registers new handlers for the given phase in the middleware stack with a custom priority.
```

Comment:
For sure. We could provide an example via documentation. Also, via `_examples`. 
Most of the documentation is not done yet. I would like to review the existent one and document in detail every package to be properly used and hacked by any dev.